pkg/cluster/cluster.go

```diff
@@ -664,3 +664,32 @@ func (c *Cluster) GetStatus() *spec.ClusterStatus {
 		Error: c.Error,
 	}
 }
+
+// ManualFailover does manual failover to a candidate pod
+func (c *Cluster) ManualFailover(curMaster *v1.Pod, candidate spec.NamespacedName) error {
+	podLabelErr := make(chan error)
+	defer close(podLabelErr)
+
+	go func() {
```

Comment:
I see no purpose in these goroutine. Patroni will do the waiting for you and only return success after new master appears in the cluster. Besides, it won't terminate if the error is returned to the main function.

Reply:
we need to be sure master has the label